envs/example/defaults.yml

```diff
@@ -312,70 +299,69 @@ keystone:
     - name: keystone
       type: identity
       description: 'Identity Service'
-      public_url: https://{{ endpoints.main }}:5001/v2.0
-      internal_url: https://{{ endpoints.main }}:5001/v2.0
-      admin_url: https://{{ endpoints.main }}:35358/v2.0
+      public_url: "{{ endpoints.keystone.url.public }}/{{ endpoints.keystone.version }}"
+      internal_url: "{{ endpoints.keystone.url.internal }}/{{ endpoints.keystone.version }}"
+      admin_url: "{{ endpoints.keystone.url.admin }}/{{ endpoints.keystone.version }}"
```

Comment:
The use of `endpoints.` is confusing here. There is an endpoints block in this same file, but it doesn't have the keys used here. It does however have keys that are also defined in the endpoints role. I don't see a need to have an `endpoints` key in this file at all, and instead define them all in the endpoints role.